### PR TITLE
Release v0.0.1 (take 4): Initial release of the compiler and SDK crates

### PR DESCRIPTION
**Pull request by greenhat**
Ref #246 

Well, the release PR against the main branch is not going to work since the VM v0.10 has MSRV 1.80 and some breaking API changes. Let's test the publishing from the next branch.

In this PR, I set the release CI job to run on merges to the `next` branch. We'll revert this after publish testing is complete.

After this PR is merged, the release CI job will publish all the unpublished crates.

### CODE REVIEW

**Review by bitwalker — APPROVED** [review]

**Comment by bitwalker** [comment]
If I understand correctly, we should merge this once #272 is green? Or should we merge it now, and it will release once we have a successful build on that branch? Feel free to merge this whenever ready if it's the latter.

**Comment by greenhat** [comment]
Since it is testing of the publishing process and not the actual public release, I thought the green test suite is not required, and we can merge it now. On the second thought, maybe publishing with a red test suite is not a good idea.
The CI job should publish the crates once it merged. Since we are not running(BTW, should we?) the tests in the `release` job, green `cargo build` is the only hard requirement.

**Pull request closed without merging**
